### PR TITLE
Test order by push down for uncollected column in join statement.

### PR DESCRIPTION
**Pull request by andreidan**

### CODE REVIEW

File: sql/src/test/java/io/crate/planner/operators/PushDownTest.java [hunk 1]
```diff
@@ -122,6 +122,22 @@ public void testOrderByOnJoinWithMultipleRelationsPushedDown() {
                                                           "]\n"));
     }
 
+    @Test
+    public void testOrderByOnJoinUncollectedColumnPushedDown() {
```
Comment (seut) [inline sql/src/test/java/io/crate/planner/operators/PushDownTest.java]: maybe better: testOrderByOnJoin**With**UncollectedColumnPushedDown

File: CHANGES.txt [hunk 2]
```diff
@@ -45,6 +45,11 @@ Breaking Changes
 Changes
 =======
 
+- Fix the order by push down optimisation for ``join`` statements that order by
```
Comment (matriv) [inline CHANGES.txt]: I would phrase it simpler for the end user: `Fixed and issue that could cause incorrect results for a ``join`` query with order by due to push down optimisation.`

File: CHANGES.txt [hunk 3]
```diff
@@ -45,6 +45,9 @@ Breaking Changes
 Changes
 =======
 
+- Fixed and issue that could cause incorrect results for a ``join`` query with
```
Comment (mfussenegger) [inline CHANGES.txt]: I thought this only affected master -> so no changes entry would be necessary as the bug isn't present in released versions?

Comment (andreidan) [reply CHANGES.txt]: @mfussenegger @matriv I thought the push down optimisation was in 2.3 already no?

Comment (matriv) [reply CHANGES.txt]: @andreidan For Union yes, not for joins...

Comment (andreidan) [reply CHANGES.txt]: @mfussenegger created https://github.com/crate/crate/pull/6921 to drop the changes and add some motivation in the LogicalPlanBuilder documentation

**Pull request merged**